client/deployment_protection.go

```diff
@@ -0,0 +1,28 @@
+package client
+
+type VercelAuthentication struct {
+	DeploymentType string `json:"deploymentType"`
+}
+
+type PasswordProtection struct {
+	DeploymentType string `json:"deploymentType"`
+}
+type PasswordProtectionWithPassword struct {
+	DeploymentType string `json:"deploymentType"`
+	Password       string `json:"password"`
+}
```

Comment:
```suggestion
type PasswordProtection struct {
	DeploymentType string `json:"deploymentType"`
	Password       *string `json:"password,omitempty"`
}
```

Reply:
Maybe, but this seems fine.

Reply:
The matching struct `PasswordProtectionWithPassword` with `tfsdk` can't use the null pointer, so I think it's easier to keep them aligned